src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -4810,10 +4810,9 @@ public static T StdDev<T>(in ReadOnlyTensorSpan<T> x)
         {
             T mean = Average(x);
             T result = T.AdditiveIdentity;
-            TensorOperation.Invoke<TensorOperation.SumOfSquaredDifferences<T>, T, T>(x, mean, ref result);
+            TensorOperation.Invoke<TensorOperation.SumOfSquaredMagnitudeDifferences<T>, T, T>(x, mean, ref result);
```

Comment:
```suggestion
            if (typeof(T) == typeof(double) || typeof(T) == typeof(float))
            {
                TensorOperation.Invoke<TensorOperation.SumOfSquaredDifferences<T>, T, T>(x, mean, ref result);
            }
            else
            {
                TensorOperation.Invoke<TensorOperation.SumOfSquaredMagnitudeDifferences<T>, T, T>(x, mean, ref result);
            }
```
Is this perhaps a viable option for your concerns on introducing the call to `Abs`? I assume the vast majority of will use this API for `double` and `float`.

Reply:
Good suggestion; thanks, @lilinus. I'll await @tannergooding's input here too.

Reply:
That would then fail for `Half` and any other user-defined IEEE 754 compatible types.

Inversely, `Abs` can be an issue for types that aren't ones complement as it can fail for `T.MinValue` (which stays `T.MinValue` after the `Abs` and so often throws). -- Although this is unlikely to be encountered in practice.

If it were to be limited, then the "correct" thing would be to use `T.IsComplexNumber` and/or `T.IsImaginaryNumber` to correctly handle such edge cases that need the additional step.

However, there isn't a way to statically determine that all `T` are complex, so it would need to be integrated into the core loop instead.

--------------------

Simply having `SquaredDifferenceOperator` be the following, as it is for TensorPrimitives, is likely fine:
```csharp
T diff = x - y;
diff = T.Abs(diff);
return diff * diff;
```

We should likely consider if `Distance` has the same issue for `Complex` numbers. I don't recall the expected behavior off the top of my head

Reply:
I talked with @tannergooding about this. Recap:

1. Distance should remain as-is, without doing the `Abs`. That matches the defined behavior for the algorithm.
2. We will skip the type checks that would avoid the `Abs` call as the `Abs` call is acceptable to keep for all scenarios (perf-wise) and the type checking would never be comprehensive anyway while it would add edge case behaviors to keep in mind
3. The `Abs` call can fail on some `MinValue`s, leading to an exception that could be thrown from `StdDev`. I will try to capture that scenario in a test and then document that an exception can be thrown from the API under the condition.